entrypoint.sh

```diff
@@ -27,4 +27,4 @@ if [[ "$RUNNER_DEBUG" == '1' ]]; then
 fi
 
 unset JAVA_HOME
-sonar-scanner $debug_flag -Dsonar.projectBaseDir=${INPUT_PROJECTBASEDIR} -Dsonar.host.url=${SONARCLOUD_URL} ${INPUT_ARGS}
+sonar-scanner "$debug_flag" -Dsonar.projectBaseDir="${INPUT_PROJECTBASEDIR}" -Dsonar.host.url="${SONARCLOUD_URL}" "${INPUT_ARGS}"
```

Comment:
Two things here:

1- Quoting the debug flag won't work here, it is one of the Exceptions mentioned in [SC2086](https://www.shellcheck.net/wiki/SC2086):

> Quoting this doesn't work, since in the default case, "$debug" would expand to one empty argument while $debug would expand into zero arguments. In this case, you can use an array with zero or one elements as outlined above, or you can use an unquoted expansion with an alternate value:

It seems doable with something like this:
```
debug=""
[[ "$RUNNER_DEBUG" == '1' ]] && debug="yes"

unset JAVA_HOME

sonar-scanner ${debug:+"--debug"} -Dsonar.projectBaseDir="${INPUT_PROJECTBASEDIR}" -Dsonar.host.url="${SONARCLOUD_URL}" ${INPUT_ARGS}
```

2- We don't want to quote the Input args right now because. We tried at some point but reverted the change for security reasons (See [revert here](https://github.com/SonarSource/sonarcloud-github-action/pull/83))

Reply:
I went for a _simple solution_ and reverted the changes in this file (cf. 447735a02a5a5477ffdbe73a2cb72ad56567ebc8). Should I try to apply your suggestion about the _debug flags_?

Thanks for putting some light on these nuances.

Reply:
Reverting these changes is good 👍 
Thank you!